tests/unit/h_periodic/views/status_test.py

```diff
@@ -0,0 +1,105 @@
+class TestStatusView:
+    # pylint: disable=too-many-arguments
+    def test_it_with_everything_ok(
+        self, tmpdir, pyramid_request, env_vars, psutil, kombu
+    ):
+        env_vars[StatusView.H_BROKER_VAR] = sentinel.h_broker
+
+        h_pid_file = tmpdir / StatusView.H_PID_FILE
+        h_pid_file.write("1234")
+
+        view = StatusView(sentinel.context, pyramid_request)
+        result = view.status()
+
+        psutil.pid_exists.assert_called_once_with(1234)
+
+        kombu.Connection.assert_called_once_with(sentinel.h_broker)
+        connection = kombu.Connection.return_value
+        connection.connect.assert_called_once_with()
+        connection.close.assert_called_once_with()
+
+        assert result == {
+            "h": {"beat": "ok", "connection": "ok", "status": "ok"},
+            "status": "ok",
+        }
+        assert view.request.response.status_int == 200
+
+    def test_it_with_no_broker_url(self, pyramid_request, env_vars):
+        env_vars[StatusView.H_BROKER_VAR] = None
+
+        result = StatusView(sentinel.context, pyramid_request).status()
+
+        assert result == {
+            "h": Any.dict.containing({"connection": "down", "status": "down"}),
+            "status": "down",
+        }
+
+    @pytest.mark.parametrize("exception", (KombuError, ConnectionRefusedError))
+    def test_it_with_bad_broker_connection(
+        self, pyramid_request, env_vars, kombu, exception
+    ):
+        env_vars[StatusView.H_BROKER_VAR] = sentinel.h_broker
+        kombu.Connection.return_value.connect.side_effect = exception
+
+        result = StatusView(sentinel.context, pyramid_request).status()
+
+        assert result == {
+            "h": Any.dict.containing({"connection": "down", "status": "down"}),
+            "status": "down",
+        }
+
+    def test_it_with_no_pid_file(self, pyramid_request, tmpdir):
+        result = StatusView(sentinel.context, pyramid_request).status()
+
+        assert result == {
+            "h": Any.dict.containing({"beat": "down", "status": "down"}),
+            "status": "down",
+        }
+
+    def test_it_with_pid_file_but_no_pid(self, pyramid_request, tmpdir, psutil):
+        h_pid_file = tmpdir / StatusView.H_PID_FILE
+        h_pid_file.write("1234")
+        psutil.pid_exists.return_value = False
+
+        result = StatusView(sentinel.context, pyramid_request).status()
+
+        assert result == {
+            "h": Any.dict.containing({"beat": "down", "status": "down"}),
+            "status": "down",
+        }
+
+    @pytest.fixture
+    def tmpdir(self, tmpdir):
+        cwd = os.getcwd()
+
+        try:  # pylint: disable=too-many-try-statements
+            os.chdir(tmpdir)
+            yield tmpdir
+        finally:
+            os.chdir(cwd)
+
+    @pytest.fixture(autouse=True)
+    def env_vars(self, patch):
+        env_vars = {}
+        environ = patch("h_periodic.views.status.os.environ")
+        environ.get = env_vars.get
+
+        return env_vars
```

Comment:
I think you need to patch the name `h_periodic.views.status.os` (the name that's actually imported into our `status` module's namespace) rather than a name in the standard `os` module's namespace. Otherwise I think you can run into confusing problems, because you've replaced `os.environ` with a mock for everyone that's importing the standard `os` module.

It's nice also to call the fixture `environ` the same as it's called in the code, rather than introduce the new name `env_vars`.

I think it's pretty easy to do:

```diff
diff --git a/tests/unit/h_periodic/views/status_test.py b/tests/unit/h_periodic/views/status_test.py
index e5349c4..8426eaa 100644
--- a/tests/unit/h_periodic/views/status_test.py
+++ b/tests/unit/h_periodic/views/status_test.py
@@ -9,8 +9,8 @@ from h_periodic.views.status import StatusView
 
 
 class TestStatusView:
-    def test_it_with_everything_ok(self, tmpdir, pyramid_request, env_vars, psutil, kombu):
-        env_vars[StatusView.H_BROKER_VAR] = sentinel.h_broker
+    def test_it_with_everything_ok(self, tmpdir, pyramid_request, os, psutil, kombu):
+        os.environ[StatusView.H_BROKER_VAR] = sentinel.h_broker
 
         h_pid_file = tmpdir / StatusView.H_PID_FILE
         h_pid_file.write('1234')
@@ -31,8 +31,8 @@ class TestStatusView:
         }
         assert view.request.response.status_int == 200
 
-    def test_it_with_no_broker_url(self, pyramid_request, env_vars):
-        env_vars[StatusView.H_BROKER_VAR] = None
+    def test_it_with_no_broker_url(self, pyramid_request, os):
+        os.environ[StatusView.H_BROKER_VAR] = None
 
         result = StatusView(sentinel.context, pyramid_request).status()
 
@@ -42,8 +42,8 @@ class TestStatusView:
         }
 
     @pytest.mark.parametrize('exception', (KombuError, ConnectionRefusedError))
-    def test_it_with_bad_broker_connection(self, pyramid_request, env_vars, kombu, exception):
-        env_vars[StatusView.H_BROKER_VAR] = sentinel.h_broker
+    def test_it_with_bad_broker_connection(self, pyramid_request, os, kombu, exception):
+        os.environ[StatusView.H_BROKER_VAR] = sentinel.h_broker
         kombu.Connection.return_value.connect.side_effect = exception
 
         result = StatusView(sentinel.context, pyramid_request).status()
@@ -84,17 +84,10 @@ class TestStatusView:
             os.chdir(cwd)
 
     @pytest.fixture(autouse=True)
-    def env_vars(self, patch):
-        env_vars = {}
-        environ = patch('h_periodic.views.status.os.environ')
-        environ.get = env_vars.get
-
-        return env_vars
-
-    @pytest.fixture(autouse=True)
-    def psutil(self, patch):
-        return patch('h_periodic.views.status.psutil')
-
+    def os(self, patch):
+        os = patch('h_periodic.views.status.os')
+        os.environ = {}
+        return os
     @pytest.fixture(autouse=True)
     def kombu(self, patch):
         return patch('h_periodic.views.status.kombu')

```

Reply:
This doesn't work for a couple of reasons:

 * You end up patching `os.path.exists` as well
 * You _really, really_ break `pytest` for some reason. It goes really nuts with errors from posix path if you patch this for some reason

Reply:
I've done this with `from os import environ` and then we can leave the rest of `os` alone